optimum/habana/transformers/models/codegen/modeling_codegen.py

```diff
@@ -343,17 +359,21 @@ def prepare_inputs_for_generation(
                 else:
                     position_ids = position_ids[:, -1]
 
+                # This `clone` call is needed to avoid recapturing cuda graphs with `torch.compile`'s  `mode="reduce-overhead`, as otherwise the input `position_ids` would have various stride during the decoding. Here, simply using `.contiguous()` is not sufficient as in the batch size = 1 case, `position_ids` is already contiguous but with varying stride which retriggers a capture.
+                position_ids = position_ids.clone(memory_format=torch.contiguous_format)
```

Comment:
add lazy_mode check

Reply:
i think clone causes perf issue, and we dont need this if it's not torch,compile,right?